### PR TITLE
feat(AWS HTTP API): Support `shouldStartNameWithService` option

### PR DESCRIPTION
**Pull request by thiagomr**
**Overview**:  
- Add a option on HTTP API to set the api gateway name starting with the service name.
- Based on Rest API implementation.
- Add unit tests on HTTP API naming function

Closes: [#9631](https://github.com/serverless/serverless/issues/9631)

### CODE REVIEW

**Review by pgrzesik — CHANGES_REQUESTED** [review]
Thank you @thiagomr - it looks good so far. We should make sure that it is included in configuration schema - otherwise it's going to be failing for users trying to use that property. Schema for `httpApi` on provider level lives here: https://github.com/serverless/serverless/blob/b775613d62ce3153084e3bff00a7db9420645be2/lib/plugins/aws/provider.js#L847

We should also add information about this property to corresponding documentation: https://github.com/serverless/serverless/blob/master/docs/providers/aws/events/http-api.md

**Comment by thiagomr** [comment]
@pgrzesik great, I'll make these changes. A quick question, is there any documentation on this type of change (changing resource options requires update provider, for example)? I'm curious because I tested this and it worked (even without updating the provider).

**Comment by pgrzesik** [comment]
hey @thiagomr - I don't think we have it specified explicitly in the contributing docs. When you tested it, did you see validation issues that there's an unrecognized property? Because that should be the behavior and with next major it will error out by default instead of just printing a warning.

**Comment by thiagomr** [comment]
@pgrzesik No, I didn't see any validation error. It has been successfully deployed. I'll take the opportunity to check this behavior. Please let me know if you think it's a good idea to add some documentation about this kind of contribution (since changing resource properties should be something that happens frequently)

**Comment by pgrzesik** [comment]
Hello @thiagomr, I've just tested your branch locally with the following configuration:

```
service: example-function

provider:
  name: aws
  runtime: nodejs12.x
  httpApi:
    shouldStartNameWithService: true

functions:
  exampleFunction:
    handler: src/handler.exampleFunction
    events:
      - httpApi: '*'
```

and I've got the following warning (that can also be an error for people with strict validation which will be the default starting with new major). 

validation

The reason for the above is that the property is not a part of schema as I linked in one of previous comments. 

It's a great idea about adding notes about that - I'll see where we could fit it :+1:

**Comment by thiagomr** [comment]
@pgrzesik I got it, I see it now. Thanks for the update, I'll send the requested changes soon!

**Comment by pgrzesik** [comment]
Thank you @thiagomr :bow: Please re-request for a review whenever you feel ready :raised_hands:

**Comment by thiagomr** [comment]
Hey @pgrzesik, I've submitted the requested changes. Please let me know if it looks good to you : )

File: lib/plugins/aws/provider.js [hunk 1]
```diff
@@ -924,6 +924,7 @@ class AwsProvider {
                 metrics: { type: 'boolean' },
                 useProviderTags: { const: true },
                 disableDefaultEndpoint: { type: 'boolean' },
+                shouldStartNameWithService: { type: 'boolean' },
```
Comment (pgrzesik) [inline lib/plugins/aws/provider.js:927]: I think it should be `const: true` instead as setting it to `false` does not make sense, so the only reasonable value is `true`

**Review by pgrzesik — CHANGES_REQUESTED** [review]
thank you @thiagomr :bow: Changes looks great, I only have one minor suggestion, please let me know what do you think :bow:

File: lib/plugins/aws/provider.js (hunk 1 above)
Comment (thiagomr) [reply lib/plugins/aws/provider.js]: Totally agree. I've already updated it. Thanks!

File: docs/providers/aws/events/http-api.md [hunk 2]
```diff
@@ -443,3 +443,13 @@ provider:
   httpApi:
     disableDefaultEndpoint: true
 ```
+
+### Service Naming
+
+You can use the `${service}-${stage}` naming for HTTP Gateway instead the default option `${stage}-${service}`.
```
Comment (pgrzesik) [inline docs/providers/aws/events/http-api.md:449]: Let's rewrite this line as

```
You can use the `shouldStartNameWithService` option to change the naming scheme for HTTP API from the default `${stage}-${service}` to `${service}-${stage}`.
```

**Review by pgrzesik — COMMENTED** [review]
Thank you @thiagomr - I'm sorry that I've missed one thing in my previous review. It should be the last thing and we'll be ready to merge, thank you :bow:

File: docs/providers/aws/events/http-api.md (hunk 2 above)
Comment (thiagomr) [reply docs/providers/aws/events/http-api.md]: No worries, this is clearer. It's done, thanks!

**Review by pgrzesik — APPROVED** [review]
Thank you @thiagomr, it looks great :+1:

**Pull request merged**